osvr/RenderKit/RenderManagerBase.cpp

```diff
@@ -1433,188 +1463,407 @@ namespace renderkit {
         /// @todo Replace all of the quatlib math with Eigen math below,
         /// or with direct calls to the core library.
 
-        /// We need to determine the transformation that takes points
-        /// in the space we're going to render in and moves them into
-        /// the space described by the projection matrix.  That space
-        /// is eye space, which has the eye at the origin, X to the
-        /// right, Y up, and Z pointing into the camera (opposite to
-        /// the viewing direction).
-
-        /// Include the impact of rotating the screen around the
-        // eye location for HMDs who have this feature.  This is
-        // computed in terms of the percent overlap of the screen.
-        // We rotate each eye away from the other by half of the
-        // amount they should not overlap.  NOTE: This assumes
-        // that both eyes are at the same location w.r.t. the
-        // overlap percent.
-        // @todo Verify this assumption.
-        q_xyz_quat_type q_rotatedEyeFromEye;
-        makeIdentity(q_rotatedEyeFromEye);
-        double rotateEyesApart = 0;
-        double overlapFrac =
-            m_params.m_displayConfiguration->getOverlapPercent();
-        if (overlapFrac < 1.) {
-            const auto hfov =
-                m_params.m_displayConfiguration->getHorizontalFOV();
-            const auto angularOverlap = hfov * overlapFrac;
-            rotateEyesApart = util::getDegrees((hfov - angularOverlap) / 2.);
-        }
-        // Right eyes should rotate the other way.
-        if (whichEye % 2 != 0) {
-            rotateEyesApart *= -1;
-        }
-        rotateEyesApart = osvr::common::degreesToRadians(rotateEyesApart);
-        q_from_axis_angle(q_rotatedEyeFromEye.quat, 0, 1, 0, rotateEyesApart);
-
-        /// Include the impact of the eyeFromHead matrix.
-        // This is a translation along the X axis in head space by
-        // the IPD, or its negation, depending on the eye.
-        // We assume that even eyes are left eyes and odd eyes are
-        // right eyes.  We further assume that head space is between
-        // the two eyes.  If the display descriptor wants us to swap
-        // eyes, we do so by inverting the offset for each eye.
-        /// @todo dealing with eye tracking here or mono displays
-        q_xyz_quat_type q_headFromRotatedEye;
```

Comment:
I recommend just doing the if(eyes) around this block, and selecting the appropriate offset for whichever eye is current by subtracting it from head space here and putting it into the .xyz part of q_headFromRotatedEye.  This leaves both code paths the same except where they need to diverge and will avoid code drift like that found below where the eye rotation has gone missing on one path.

Reply:
Thanks for the review. I added a commit which I think fits what you're describing here. What about the rotation for each eye pose? Should eye poses always be a translation in headspace, rotating with the head pose, or can each eye rotate independently?

Reply:
The eye rotations in this case aren't meant to handle tracking which way the eyes are looking, but rather which direction the screens are with respect to the eyes, to handle cases where the displays are mounted at an angle.  The diff seems to still be showing a monolithic replacement of the code, so I'll need to take another look through the whole thing.